modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+                {username, ResponseCodeConstants.CODE_NOT_IMPLEMENTED},
+                {username, ResponseCodeConstants.CODE_BAD_REQUEST}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteFails")
+    public void testDeleteFails(String userName, int expectedScimResponseStatus)
+            throws NotFoundException, NotImplementedException, BadRequestException, CharonException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+
+        if (expectedScimResponseStatus == ResponseCodeConstants.CODE_INTERNAL_ERROR) {
+
+            when(AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+
+            SCIMResponse outputScimResponse = meResourceManager.delete(userName, null);
+            Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        } else if (expectedScimResponseStatus == ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND) {
+
+            doThrow(new NotFoundException()).when(userManager).deleteMe(userName);
+
+            when(AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+
+            SCIMResponse outputScimResponse = meResourceManager.delete(userName, userManager);
+            Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        } else if (expectedScimResponseStatus == ResponseCodeConstants.CODE_NOT_IMPLEMENTED) {
+
+            doThrow(new NotImplementedException()).when(userManager).deleteMe(userName);
+
+            when(AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new NotImplementedException()));
+
+            SCIMResponse outputScimResponse = meResourceManager.delete(userName, userManager);
+            Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        } else if (expectedScimResponseStatus == ResponseCodeConstants.CODE_BAD_REQUEST) {
+
+            doThrow(new BadRequestException()).when(userManager).deleteMe(userName);
+
+            when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+
+            SCIMResponse outputScimResponse = meResourceManager.delete(userName, userManager);
+            Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        }
+
+    }
+
+    @DataProvider(name = "dataForTestDeleteFailsINCharonExceptionOnly")
+    public Object[][] dataToTestDeleteCharonExceptionOnly()
```

Comment:
```suggestion
    public Object[][] dataToTestDeleteUserCharonException()
```

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d